code/modules/mod/mod_ai.dm

```diff
@@ -0,0 +1,77 @@
+/obj/item/mod/control/transfer_ai(interaction, mob/user, mob/living/silicon/ai/intAI, obj/item/aicard/card)
+	. = ..()
+	if(!.)
+		return
+	if(!open) //mod must be open
+		to_chat(user, "<span class='warning'>[src] must be open in order to allow a transfer.</span>")
+		return
+	switch(interaction)
+		if(AI_TRANS_TO_CARD)
+			if(!ai)
+				to_chat(user, "<span class='warning'>No AI detected in [src].</span>")
+				return
+			if(!do_after(user, 5 SECONDS, target = src))
+				return
+			intAI = ai
+			intAI.ai_restore_power()//So the AI initially has power.
+			intAI.control_disabled = TRUE
+			intAI.radio_enabled = FALSE
+			intAI.disconnect_shell()
+			intAI.forceMove(card)
+			card.AI = intAI
+			for(var/datum/action/action as anything in actions)
+				if(action.owner == intAI)
+					action.Remove(intAI)
+					qdel(action)
```

Comment:
can this remove a malf AI's action buttons?

Reply:
research